src/unitxt/eval_assist_llm_as_judge_direct.py

```diff
@@ -96,12 +96,28 @@ def get_criterias(self, task_data, eval_count):
                     except:
                         raise Exception(
                             "The criteria dict couldn't be parsed correctly. It has to have the json representation of CriteriaWithOptions"
-                        )
+                        ) from None
                 elif isinstance(task_data_criteria, str):
-                    criteria = fetch_artifact(task_data_criteria)[0]
+                    try:
+                        criteria = fetch_artifact(task_data_criteria)[0]
+                    except Exception:
```

Comment:
This seems a bit error prone.  If I mistype the name of a catalog asset, it will become a textual criteria and get weird results.  It will also have performance issues, searching for the artificat.  
I think this option need to be more explicit.

Reply:
For example, have different field names for each use case.

Reply:
I have a better suggestion. Keep the code as is, but add a field operator that takes a string and convert it into a dict. So if the raw data has a string creteria it will convert it.  There could also  be multiple diffeentoperator.  One that conerys to Yes/No. Another that converts to Yes/no/partially.

Reply:
I didn't fully understand your proposed field operator? Can you write some pseudo-code for the `process_value` method?

Just to be in the same page, these are the source and types of criterias we could potentially accept in a direct assessment evaluation. We can narrow the list.
- If it is a single criteria it can be set in the constructor:
  - a catalog criteria string if the evaluator is specified as a catalog string (square brackets). Still to check if a dict representation would work here.
  - any criteria representation if the evaluator is created using code (not from catalog), like a catalog criteria string or a `CriteriaWithOptions` object.
- If multiple criterias, they have to be specified in the `criteria` field of the `input_fields`. It can be:
  - a string representing a criteria in the catalog,
  - a string representing the description of the criteria with a default set of `options` and `option_map` with empty respective description (e.g. Yes/No, Yes/No/Partially)
  - a string of the json representation of the criteria, e.g `"{'description': '', 'options': [{'...'}, {'...'}, {'...'}], 'option_map':{...}}"`
  - a dict: the json representation of the criteria
  - A `CriteriaWithOptions` object, no parsing needed.

Reply:
I suggest there will be different operators to convert the input in the original raw data field to the Critera.

class LoadCriteria(FieldOperator):
    def process_value(self, text: Any) -> Any:
         return   fetch_artifact(task_data_criteria)[0]
        
class CreateCriteriaFromDict(FieldOperator):
     def process_value(self, text: Any) -> Any:
         criteria = CriteriaWithOptions(
                             name=task_data_criteria["name"],
                             description=text["description"],
                             options=[
                                 CriteriaOption(
                                     name=option_dict["name"],
                                     description=option_dict["description"],
                                 )
                                 for option_dict in text["options"]
                             ],
           return criteria                  
           
class CreateCriteriaFromString(FieldOperator):
     def process_value(self, text: Any) -> Any:
            criteria = CriteriaWithOptions(
                             name=f"Unknown ({text[:20]}...)",
                             description=text,
                             options=[
                                 CriteriaOption(name="Yes", description=""),
                                 CriteriaOption(name="No", description=""),
                             ],
                             option_map={
                                 "Yes": 1.0,
                                 "No": 0.0,
                             },
                         )
           return criteria
           
Then:
    "test": [
         {"question": "How is the weather?"},
         {"judgement": "Response should discussthe weather, and only the weather")}
     ]
card = TaskCard(
     loader=LoadFromDictionary(data=data, data_classification_policy=["public"]),
     preprocess_steps=[
                                    CreateCriteriaFromString(from_field="judgement", to_field="criteria"),
                                    
                                    ],
     task=Task(
         input_fields={"question": str, "criteria": Criteria},
         reference_fields={},
         prediction_type=str,
         metrics=[
             "metrics.llm_as_judge.eval_assist.direct_assessment.rits.llama3_1_70b[context_fields=question]"
         ],
     ),
 )

Reply:
We are checking if this can work - I'll send an example if it does.

Reply:
I've implemented the above.

See examples/evaluate_eval_assist_direct_criteria_from_dataset.py

It seems to work, but I think it exposed another bug. There returned score is '1' although the judge selected the 'No' options:

score:
    instance:
        score (int):
            1
        llm_as_a_judge_score (int):
            1
        positional_bias (bool):
            False
        selected_option (str):
            No
        positional_bias_selected_option (str):
            No
        assessment (str):
            No
            
            The response is not funny as it's a weak play on words, and the punchline "To cat to the other side" is a
            n obvious and unoriginal variation of the classic "To get to the other side" phrase.
        positional_bias_assessment (str):
            No

Reply:
Thanks for contributing this @yoavkatz . The bug is related to the fact that the `option_map` attribute was not being correctly set and the default value for the `option_map` was 1. Do you think the `option_map` should be required? It is currently optional. For simple usage, it makes sense to keep it as optional, we could set default float values for the options.

The criteria in the `task_data` objects is currently a dict (with `__type__` key) although a `CriteriaWithOptions` object is returned by `CreateYesNoCriteriaFromString`. So, we have to call `fetch_artifact()`. Is this expected?

Reply:
> For simple usage, it makes sense to keep it as optional, we could set default float values for the options.

How could we give defaults to option_map? Don't we need to set it always to convert it.

I will check regarding desearlizing the object in the task dat - for now - try to fetch_artificat().

Reply:
To give defaults to `option_map` we could assume that the options are an ordinal variable in crescending order and assign the value of `i/n` to the _ith_ option, where `n` is the number of options. If we make the `option_map` required, which values should a user set to a nominal categorical variable, where options aren't ordered?

Reply:
I see your point.  I think that if the options are   "Happy" , "Sad", "Angry" we create should scores like "happy_score" and have it 1, if the chosen category is "happy" or else otherwise.    This will allow to aggregate over these score, over all instances (if we use the same criteria).

Now the question what will be the default "score" field.  If option_map is not set, then it could be assign the value of i/n to the ith option, where n is the number of options.   If option_map  is set, it should be used to map.